src/api/app/models/user.rb

```diff
@@ -218,29 +218,39 @@ def self.find_with_credentials_via_ldap(login, password)
 
   # The currently logged in user (might be nil). It's reset after
   # every request and normally set during authentification
+  # DEPRECATED! Prefer possibly_nobody or required
   def self.current
     Thread.current[:user]
   end
 
-  def self.nobody
-    Thread.current[:nobody] ||= find_nobody!
-  end
-
   # Currently logged in user or nobody user if there is no user logged in.
   # Use this to check permissions, but don't treat it as logged in user. Check
   # is_nobody? on the returned object
-  def self.current_or_nobody
+  def self.possibly_nobody
     current || nobody
   end
 
-  def self.session=(user)
-    Thread.current[:user] = user
+  # Currently logged in user. Will thrown an exception if no user is logged in.
+  # So the controller needs to require login if using this (or models using it)
+  def self.required
+    raise ArgumentError, 'Requiring user, but found nobody' unless not_required
+    current
+  end
+
+  # Currently logged in user or nil
+  def self.not_required
```

Comment:
I had second thoughts on these 2. I would by now prefer `User.session` for `User.not_required` and `User.session!` for `User.required`.